### PR TITLE
Fix error if object destructuring in require-dollar-before-jquery

### PR DESCRIPTION
**Pull request by SimenB**
#1373

I have no idea how to do this. I left a comment in the file. Could I get some help? :smile:

### CODE REVIEW

File: lib/rules/require-dollar-before-jquery-assignment.js [hunk 1]
```diff
@@ -76,9 +76,15 @@ module.exports.prototype = {
...
-                varName = left.name;
-                right = token.init;
+                if (token.id.type === 'Identifier') {
+                    left = token.id;
+                    varName = left.name;
+                    right = token.init;
+                } else { // token.id.type === 'ObjectPattern'
+                    // What do I check here?
+                    // token.id.properties is an array both times during the test.
+                    // How to I check first on, then the other?
```
Comment (SimenB) [inline lib/rules/require-dollar-before-jquery-assignment.js]: I could return maybe, you don't use destructuring for jQuery assignments, do you?

**Comment by mrjoelkemp** [comment]
I just did something like: 

image

In regards to the use within destructuring, in the case of

```js
var {foo} = {foo: $('.foo')};
```

The validation would occur on the right have side.

If it were:

```js
var {foo} = {$foo: $('.foo')};
```

Then the destructuring is invalid since the destructured var would need to be `$foo`.

As such, simply returning is viable.

Would you like to make those adjustments to the PR?

**Comment by SimenB** [comment]
That I can do! I'll do it after work

**Comment by SimenB** [comment]
@mrjoelkemp How about now? :smile:

**Comment by mrjoelkemp** [comment]
LGTM

**Comment by mrjoelkemp** [comment]
We'll fix the commit message before merge. Thanks for contributing!

**Comment by SimenB** [comment]
Oh, that one I forgot, I just squashed away. I'll give the msg a whack

**Comment by markelog** [comment]
Thank you

**Pull request closed without merging**